### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by KateOrient**

### CODE REVIEW

File: task4/script.js [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+        ]
+    }];
+
+//- Module
+let photoActions = (function () {
+    let getPhotoPosts = (skip, top, filterConfig = {}) => {
+        skip = skip ? skip : 0;
+        top = top ? top : 10;
+        let resPhotoPosts = photoPosts;
+        if (filterConfig.length !== 0) {
```
Comment (YanaYaroshevich) [inline task4/script.js:353]: length?

File: task4/script.js [hunk 2]
```diff
@@ -0,0 +1,702 @@
...
+                });
+            }
+        }
+        return resPhotoPosts.slice(skip, top + skip);
+    };
+
+    let getPhotoPost = id => {
+        let res = null;
+        photoPosts.forEach(post => {
+            if (post.id === id) {
```
Comment (YanaYaroshevich) [inline task4/script.js:381]: find

File: task4/script.js [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+
+    let validateHashTags = post => {
+        if ('hashTags' in post) {
+            if ({}.toString.call(post.hashTags) !== '[object Array]') {
+                console.log('Field \'hashTags\' has an invalid format');
+                return false;
+            }
+            let regexp = new RegExp('#[A-Za-z_]+');
+            let isHashTag = true;
+            post.hashTags.forEach(hashtag => isHashTag = regexp.test(hashtag) ? isHashTag : false);
```
Comment (YanaYaroshevich) [inline task4/script.js:466]: nice!

Comment (YanaYaroshevich) [reply task4/script.js:466]: use every

File: task4/script.js [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+            return true;
+        }
+        return false;
+    };
+
+    let editPhotoPost = (id, editedPost) => {
+        let post = getPhotoPost(id);
+        if (post !== null) {
+            let res = true;
+            Object.keys(editedPost).forEach(prop => res = prop in post ? res : false);
```
Comment (YanaYaroshevich) [inline task4/script.js:516]: every?

File: task4/script.js [hunk 5]
```diff
@@ -0,0 +1,702 @@
...
+            if (res) {
+                if (validatePhotoPost(editedPost, false)) {
+                    photoPosts.splice(photoPosts.indexOf(post), 1);
+                    photoPosts.push({...post, ...editedPost});
+                    photoPosts = photoPosts.sort((a, b) => a.createdAt - b.createdAt);
```
Comment (YanaYaroshevich) [inline task4/script.js:521]: wrong compare func

File: task4/script.js (hunk 1 above)
Comment (KateOrient) [reply task4/script.js:353]: fixed. or it's better to remove this check at all?

File: task4/script.js (hunk 2 above)
Comment (KateOrient) [reply task4/script.js:381]: fixed

File: task4/script.js (hunk 3 above)
Comment (KateOrient) [reply task4/script.js:466]: fixed

File: task4/script.js (hunk 5 above)
Comment (KateOrient) [reply task4/script.js:521]: fixed

File: task4/script.js (hunk 4 above)
Comment (KateOrient) [reply task4/script.js:516]: fixed

**Pull request merged**